src/clinical/service-worker-thread/tasks.ts

```diff
@@ -213,6 +216,35 @@ const mapEntityDocuments = (
   };
 };
 
+export function filterDonorIdDataFromSearch(donors: Donor[], query: ClinicalSearchQuery) {
+  const { donorIds, submitterDonorIds } = query;
+
+  const useFilteredDonors =
+    (donorIds && donorIds.length) || (submitterDonorIds && submitterDonorIds.length);
```

Comment:
checking using a util function like `isEmpty` makes this line cleaner. right now `useFilteredDonors` evaluates to a `number` type rather than a `boolean`